NumLib/ODESolver/NonlinearSolver.h

```diff
@@ -122,6 +122,7 @@ class NonlinearSolver<Matrix, Vector, NonlinearSolverTag::Newton> final
     std::size_t _res_id = 0u;           //!< ID of the residual vector.
     std::size_t _J_id = 0u;             //!< ID of the Jacobian matrix.
     std::size_t _minus_delta_x_id = 0u; //!< ID of the \f$ -\Delta x\f$ vector.
+    std::size_t _x_new_id = 0u;         //!< ID of the vector storing \f$ x - (-\Delta x) \f$.
```

Comment:
Sorry, I don't understand the question. The use of the variable is up to the one who implements a process.
For `postIteration()` it is natural to implement some checks.
I don't have a good example for `preIteration()`. But you could precompute some cached values or initialize something, e.g. external software.
Passing `x` to `preIteration()` probably overlaps pretty much with what can be done in the `assemble...()` call.